test/functional/gridfs.test.js

```diff
@@ -3884,4 +3884,29 @@ describe('GridFS', function() {
       });
     }
   });
+
+  /**
+   * @ignore
+   */
+  it('should correctly create an index', function(done) {
+    const { configuration } = this;
+    const client = configuration.newClient();
+
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const db = client.db(configuration.db);
+      const gridStore = new GridStore(db, 'test_gs_save_empty_file', 'w');
+      gridStore.open(err => {
+        expect(err).to.not.exist;
+        db.collection('fs.files').createIndex({ filename: 1, uploadDate: 1 }, {}, (err, val) => {
```

Comment:
out of curiosity, did the code require you to pass in an empty object for options here? You should be able to omit it:
```
 db.collection('fs.files').createIndex({ filename: 1, uploadDate: 1 }, (err, val) => {
```